### ISSUE TITLE
Crash during reporting of ConfigurationError

### ISSUE THREAD

**Issue by wosc** [flyingcircusio/batou#75]
```
File ".../batou/remote_core.py", line 375, in <module>
      TypeError: '<' not supported between instances of 'str' and 'int'
```

The line in question is `deployment.environment.exceptions.sort(key=lambda x: x.sort_key)`. Python-3 is much stricter than Python-2 with comparisons (I've seen this same issue during other porting efforts rather frequently).

(As for my concrete case, after locally commenting out the offending line, I got a jumbled mess that boiled down to "Overrides for undefined attributes", which then was easily resolveable.)

**Comment by wosc** [comment]
I've tried adjusting this in #76 but this seems to jumble the actual sorting somewhat unpredictably, so now I'm not sure if that naive idea is actually feasible.

**Comment by ctheune** [comment]
Right. This shouldn't be necessary. We 'just' have to be careful that the the types of the order hierarchy stay compatible for comparison within the subtree.

**Comment by ctheune** [comment]
The reordering in #76 is likely because numerical sort is moved to quasi-lexicographic sort which produces numerical sort because it screws up 'endianess'...

**Comment by ctheune** [comment]
Weird. I can't reproduce this error and I wrote a test suite that runs comparisons between all possible combinations. Any chance you can get me the 'garbage' that you experienced?

**Comment by wosc** [comment]
Riiiight. I can't reproduce the error with batou-2.0 either. Checking back, we were using 2.0b11 at the time, for which I'll add the minimal reproduction below, just for completeness. But after looking at the changelog, _now_ I'm stumbling upon #52 (which was included in 2.0b12), which I probably should have found before reporting this issue (again), but somehow I didn't see that one. Sorry for the noise.


```
class Component1(Component):
    def configure(self):
        self.provide('component1')

class Component2(Component):
    def configure(self):
        self.require_one('component2', self.host)

###
[environment]
connect_method = local
[hosts]
localhost = component1, component2
[component:component1]
nonexistent_property = value
```

**Comment by ctheune** [comment]
Right. Glad this works now and also happy about the test I managed to come up with to avoid future breakage ...
